gpt_index/indices/query/query_processor.py

```diff
@@ -0,0 +1,66 @@
+"""Query processor."""
+
+from typing import Optional
+
+from gpt_index.indices.query.schema import QueryBundle
+from gpt_index.langchain_helpers.chain_wrapper import LLMPredictor
+from gpt_index.prompts.base import Prompt
+from gpt_index.prompts.default_prompts import DEFAULT_HYDE_PROMPT
+
+
+class BaseQueryProcessor:
+    """Base class for query processor.
+
+    A query processor augments a raw query string with associated transformations
+    to improve index querying.
+    """
+
+    def __call__(self, query_str: str) -> QueryBundle:
+        """Run query processor."""
+        return QueryBundle(query_str=query_str, embedding_strs=[query_str])
+
+
+class HyDEQueryProcessor(BaseQueryProcessor):
```

Comment:
would love to have a How-To Guide + Reference section in the docs. 

reference could go in docs/reference/query.rst

Reply:
Updated reference.